source/base/qprojector.cc

```diff
@@ -1571,22 +1557,37 @@ QProjector<1>::DataSetDescriptor::subface(
 template <>
 QProjector<2>::DataSetDescriptor
 QProjector<2>::DataSetDescriptor::subface(
-  const ReferenceCell &reference_cell,
-  const unsigned int   face_no,
-  const unsigned int   subface_no,
-  const types::geometric_orientation /*combined_orientation*/,
-  const unsigned int n_quadrature_points,
+  const ReferenceCell               &reference_cell,
+  const unsigned int                 face_no,
+  const unsigned int                 subface_no,
+  const types::geometric_orientation combined_orientation,
+  const unsigned int                 n_quadrature_points,
   const internal::SubfaceCase<2>)
 {
-  Assert(reference_cell == ReferenceCells::Quadrilateral, ExcNotImplemented());
-  (void)reference_cell;
-
-  Assert(face_no < GeometryInfo<2>::faces_per_cell, ExcInternalError());
-  Assert(subface_no < GeometryInfo<2>::max_children_per_face,
+  Assert(face_no < reference_cell.n_faces(), ExcInternalError());
+  Assert(subface_no <
+           reference_cell.face_reference_cell(0).n_isotropic_children(),
          ExcInternalError());
 
-  return ((face_no * GeometryInfo<2>::max_children_per_face + subface_no) *
-          n_quadrature_points);
+  const unsigned int n_subfaces =
+    reference_cell.face_reference_cell(0).n_isotropic_children();
+
+  const unsigned int orientation_offset =
+    combined_orientation == ReferenceCell::default_combined_face_orientation() ?
+      1 :
+      0;
+
+  if (reference_cell == ReferenceCells::Triangle)
+    return (2 * face_no * n_subfaces + n_subfaces * orientation_offset +
+            subface_no) *
+           n_quadrature_points;
+  else if (reference_cell == ReferenceCells::Quadrilateral)
+    return ((1 - orientation_offset) * reference_cell.n_faces() * n_subfaces +
+            (face_no * n_subfaces + subface_no)) *
+           n_quadrature_points;
+  else
+    DEAL_II_ASSERT_UNREACHABLE();
+  return numbers::invalid_unsigned_int;
```

Comment:
This is basically done but we can simplify it now that 0 is the default orientation (the type of reference cell shouldn't matter).